.github/workflows/ci.yml

```diff
@@ -157,13 +157,40 @@ jobs:
           AGENT_MODE: ${{ vars.AGENT_MODE }}
           AGENT_TYPE: ${{ vars.AGENT_TYPE }}
 
+      - id: get_cassettes_diff
+        name: Detect if cassettes changed
+        if: startsWith(github.event_name, 'pull_request')
+        run: |
+          cd tests/Auto-GPT-test-cassettes
+          cassette_diff=$(git diff --name-only --diff-filter=M -- '**/*.yaml')
+          
+          if [ -n "$cassette_diff" ]; then
+            echo "Cassettes changed: $cassette_diff"
+            echo "cassettes_changed=true" >> $GITHUB_OUTPUT
+          else
+            echo "Cassettes did not change."
+            echo "cassettes_changed=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Attempt to beat Challenges if cassettes changed
+        if: startsWith(github.event_name, 'pull_request') && steps.get_cassettes_diff.outputs.cassettes_changed == 'true'
+        run: |
+          pytest -n auto --beat-challenges --cov=autogpt --cov-append --cov-branch --cov-report term-missing --cov-report xml \
+            tests/unit tests/integration tests/challenges
+          python tests/challenges/utils/build_current_score.py
```

Comment:
* Duplicate of test step
* Runs unnecessary extra tests
  * also with coverage?

Reply:
yes I will remove tests/unit tests/integration, thanks for the catch.
But yes we do want coverage. Although, since the CI runs a second time and uses the beaten challenges as regression tests, maybe it's not necessary. 
Ok based on that I think I will remove the coverage at this step and keep using the first coverage. goo catch again